x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlNativeIntegTestCase.java

```diff
@@ -156,7 +156,7 @@ protected Settings externalClusterClientSettings() {
         }
 
         Settings.Builder builder = Settings.builder();
-        builder.put("node.ml", false);
+        builder.putList("node.roles", Collections.emptyList());
```

Comment:
`node.data`, `node.master` and `node.ingest` defaulted to `true`.  So previously we were saying that the node had these roles but not `node.ml`.  Does an explicitly empty list mean "use the defaults" or "no roles"?  Either way this appears to be changing the logic:

- If an explicitly empty list means "use the defaults" then we're now defaulting the roles to `master`, `data`, `ingest`, `ml`, and `remote_cluster_client`, i.e. _including_ `ml` when we weren't before
- If an explicitly empty list means "no roles" then that's different to before because `master`, `data`, `ingest`, and `remote_cluster_client` used to default to `true`

Reply:
Yes, I tested it with an empty list and it seemed to work? I will see whats up. The original external cluster build settings seems to have all default roles set to `false`.

Reply:
@droberts195 I understood from the beginning that an empty array means "no roles". I thought this was the current behavior given how the base settings looked in `ExternalTestCluster.java` and our updates to them. It seems that "no roles" is what we want. Or at least, how I read it.

The tests obviously passed. So, 🤷

Reply:
@droberts195 tests pass if a use an empty list or if I supply master, data, and ingest.

Reply:
> tests pass if a use an empty list or if I supply master, data, and ingest

Oh, is this just for the test client?  I guess we set the roles of the nodes in the cluster somewhere else.  If it's just for the test client then no roles is probably best.

Reply:
Yeah, this is for `externalClusterClientSettings` so, it's the client :). No roles necessary.